docs/wallet/node-api.md

```diff
@@ -0,0 +1,24 @@
+---
+id: node-api
+title: node-api
+---
+
+### NodeApi
+
+The NodeApi is how the wallet project retrieves relevant node data like blocks.
+This allows the wallet for example to retrieve blocks for finding its relevant transactions.
+
+Since this is an API it can be hooked up to the `node` module of bitcoin-s but it can also be linked to
+any other implementation of your choosing. This allows you to use the bitcoin-s wallet in any schema that you
+want.
+
+The functions that the NodeApi supports are:
+
+```scala mdoc
+import org.bitcoins.core.crypto.DoubleSha256DigestBE
+
+import scala.concurrent.Future
+
+  /** Request the underlying node to download the given blocks from its peers and feed the blocks to [[org.bitcoins.node.NodeCallbacks]] */
```

Comment:
So i don't really find this useful at all for documentation purposes. How do i hook this into the wallet? What is an example of an implementation? 

For example, take a look at the chain sync documentation here: https://bitcoin-s.org/docs/next/applications/chain#syncing-and-verifying-block-headers

 and the new filter sync document i've added in #1209